### ISSUE TITLE
Shape key from surface mesh not influing on Guide curves position

### ISSUE THREAD

**Issue by mathieuauvray** [ikakupa/medusa-nodes-bug-reports#1]
Hey there !

I've noticed that when the surface mesh has shapekeys, the generated hairs do move with the shape keys.
But not the combed hair guides... Resulting in poor result with the interpolation.

(I'm testing this with fur on a character's face with multiple expressions...)

Is there a work around ?
Thanks a lot !
Mathieu

**Comment by ikakupa** [comment]
Hello Mathieu,

Yes, the guides doesn't move at the moment and it is indeed unwanted behavior.

This would be a work around:
If we add the **Deform Curves on Surface** node to the **Generator** node group it looks better, and the guides seem to follow the sculp surface with the shape keys.


https://user-images.githubusercontent.com/64023824/216660378-ef50fca3-a1d7-4a46-8edc-f2df81d743b0.mp4
I am going to do some testing to find out if this node will bring some noticeable performance penalty. 

Thanks for submitting this.
Irakli

**Comment by mathieuauvray** [comment]
You're very welcome !
Let me know if that's possible. And if it's too damaging for the performance, maybe there could be a switch to turn it off/on, when needed ?

Regarding the use of your amazing addon for fur, I have an idea to submit to you to help with the combing process. Is there a place where I can send you this suggestion ?

**Comment by ikakupa** [comment]
Yes, I will let you know. Probably we need another deformer for animation or simulation.

There is another template for feature request, you can use it if you have some ideas you would like to share.
